### PR TITLE
Kotlin multiplatform mobile

### PR DESCRIPTION
**Pull request by StefMa**
This PR is the starter code for #10 

It will prepare the whole project using Kotlin Multiplatform Mobile.
But for now it will not bring *any features* of KMM.
This means I simply moved all related code to the Multiplatform `android` part.
Nothing more.

Beside of that it will also change how to consume this library (in pure Android projects).
So it will likely a breaking change release (if you want to merge this and still publish via jitpack.io).

You can compare [this jitpack output](https://jitpack.io/com/github/stefma/track/0.4.0/build.log) (and generated artifacts) with [this jitpack output](https://jitpack.io/com/github/stefma/track/kotlin-multiplatform-mobile-6e97d671a5-1/build.log).

The still existing "android only" artifact can be consumed like the following
> com.github.stefma.track:track:kotlin-multiplatform-mobile-SNAPSHOT

(Because of the fork of course. Should be `com.github.gustavkarlsson.track:track:[GIT_TAG]` if you merge it).

The Kotlin Multiplatform Android *Release* version can be consumed like
> com.github.stefma.track:lib-libandroid:kotlin-multiplatform-mobile-SNAPSHOT

(Because of the fork of course. Should be `com.github.gustavkarlsson.track:lib-libandroid:[GIT_TAG]` if you merge it).

But, the first one will not work anymore as soon as you add something to `common`. 
Then only the second version would work.
But also only for `release` builds. Not for `debug` builds.
If we want to support both buildTypes if either have to publish both versions (which have an different artifact name as I remember correctly) *or* Android consumers has to declare a "matchingFallbacks" in their buildTypes...

### CODE REVIEW

File: lib/src/libAndroidTest/kotlin/se/gustavkarlsson/track/sqlite/SqliteTrackTest.kt [hunk 1]
```diff
@@ -47,7 +47,7 @@ class SqliteTrackTest {
     )
 
     @Test
-    fun `get no record`() {
+    fun get_no_record() {
```
Comment (gustavkarlsson) [inline lib/src/libAndroidTest/kotlin/se/gustavkarlsson/track/sqlite/SqliteTrackTest.kt:50]: Why the underscores?

File: lib/build.gradle.kts [hunk 2]
```diff
@@ -26,6 +61,8 @@ task<Jar>("sourcesJar") {
     archiveClassifier.set("sources")
 }
 
+group = "com.github.gustavkarlsson"
+version = "0.4.1"
```
Comment (gustavkarlsson) [inline lib/build.gradle.kts:65]: Why add these?

**Review by gustavkarlsson — CHANGES_REQUESTED** [review]
Looks cool! Too bad about the new directory structure... Kind of shows that kmp is STILL a bit messy.

Need to take another look before I merge it though. Also not sure about merging anything to master until the whole thing is there... Maybe I should create a new multiplatform branch for the "epic"... What do you think?

`lib/src/libAndroidAndroidTest/.DS_Store` can be deleted

**Comment by StefMa** [comment]
> Maybe I should create a new multiplatform branch for the "epic"

Hmm.. not sure about this 🤔 
Because state *today* we have a working Android solution.
If we decide we go the second way (`com.github.stefma.track:lib-libandroid:kotlin-multiplatform-mobile-SNAPSHOT`)  nothing will be changed on that in the future for existing Android projects.
but is of course your choice 😇 

Let me clean up this PR even more (adjust GH Actions and Readme) to make stuff more clear.

File: lib/build.gradle.kts (hunk 2 above)
Comment (StefMa) [reply lib/build.gradle.kts:65]: This is required to publish a KMM project.
See https://kotlinlang.org/docs/reference/mpp-publish-lib.html#publish-a-multiplatform-library

In order words:
If you would publish the library to bintray (for example) **this** would be the group name and version.
So we can basically use whatever you like here. E.g. `se.gustavkarlsson.track:lib-libandroid:1.0.0`

File: lib/src/libAndroidTest/kotlin/se/gustavkarlsson/track/sqlite/SqliteTrackTest.kt (hunk 1 above)
Comment (StefMa) [reply lib/src/libAndroidTest/kotlin/se/gustavkarlsson/track/sqlite/SqliteTrackTest.kt:50]: Android tests (which are compiled to the dex format) can't handle names with the ` symbol.
For some reasons the Kotlin compile complains about this, *even* if this is not an Android but an unit test :(

File: lib/build.gradle.kts (hunk 2 above)
Comment (gustavkarlsson) [reply lib/build.gradle.kts:65]: I know what they are used for in the general sense, but I don't use any traditional publishing method which would require it. I use jitpack and it already infers group, id and version.

**Comment by gustavkarlsson** [comment]
> > Maybe I should create a new multiplatform branch for the "epic"
> 
> Hmm.. not sure about this 🤔
> Because state _today_ we have a working Android solution.
> If we decide we go the second way (`com.github.stefma.track:lib-libandroid:kotlin-multiplatform-mobile-SNAPSHOT`) nothing will be changed on that in the future for existing Android projects.
> but is of course your choice 😇
> 
> Let me clean up this PR even more (adjust GH Actions and Readme) to make stuff more clear.

I know the state today works (kudos!), but I don't want to merge to master until it offers some actual improvement, and that only comes after another platform is supported.

**Comment by StefMa** [comment]
> I know the state today works (kudos!), but I don't want to merge to master until it offers some actual improvement, and that only comes after another platform is supported.

Ok, understand.
As discussed yesterday it make sense to take a look at [SQLDelight](https://github.com/cashapp/sqldelight).
I will try my best to implement it 🙂

**Comment by StefMa** [comment]
If you are still interested in that, pleas let me know.

**Pull request closed without merging**